### DISCUSSION TITLE
commerceCurrency Twig filter - choose formatting language

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/commerce#2470] [Ideas]
Currently, this filter chooses formatting setting in control panel or current site locale. It would be nice if there was additional option passed to this filter that would allow overwrite it.

**Comment by nfourtythree** [comment]
Hi @piotrpog 

The `commerceCurrency` twig filter uses the `formatAsCurrency` method from the [Currency helper](https://github.com/craftcms/commerce/blob/develop/src/helpers/Currency.php#L65-L107) under the hood.

This has parameters for passing in a different currency to format that amount in. So as long as you have those currencies set up in `Commerce > Store Settings > Payment Currencies` you will be able to have code like the following:

```twig
{# base currency is USD #}
{% set amount = 123.99 %}
<pre>{{ dump(amount|commerceCurrency('GBP', true)) }}</pre>
<pre>{{ dump(amount|commerceCurrency('EUR', true)) }}</pre>
```

Hope this helps, thanks!
